cob_cartesian_controller/include/cob_cartesian_controller/cartesian_controller.h

```diff
@@ -32,56 +32,59 @@
 #include <ros/ros.h>
 #include <vector>
 #include <string.h>
-#include <cob_cartesian_controller/CartesianControllerAction.h>
+#include <boost/shared_ptr.hpp>
+
+#include <tf/transform_listener.h>
+#include <tf/transform_broadcaster.h>
+#include <tf/transform_datatypes.h>
+
 #include <actionlib/server/simple_action_server.h>
+#include <cob_cartesian_controller/CartesianControllerAction.h>
 
-#include <cob_cartesian_controller/helper_classes/data_structures.h>
-#include <cob_cartesian_controller/helper_classes/utils.h>
+#include <cob_cartesian_controller/trajectory_interpolator/trajectory_interpolator.h>
+#include <cob_cartesian_controller/cartesian_controller_data_types.h>
+#include <cob_cartesian_controller/cartesian_controller_utils.h>
 
 typedef actionlib::SimpleActionServer<cob_cartesian_controller::CartesianControllerAction> SAS_CartesianControllerAction_t;
 
 class CartesianController
 {
 public:
-    void run();
     bool initialize();
     
     // Main functions
-    void posePathBroadcaster(std::vector<geometry_msgs::Pose>& pose_vector);
-    void movePTP(geometry_msgs::Pose target_pose, double epsilon);
-    void holdPosition(geometry_msgs::Pose pose);
+    bool posePathBroadcaster(const geometry_msgs::PoseArray& cartesian_path);
+    bool movePTP(geometry_msgs::Pose target_pose, double epsilon);
     
     // Helper function
-    void timerCallback(const ros::TimerEvent& event);
-    void startTracking();
-    void stopTracking();
+    bool startTracking();
+    bool stopTracking();
 
     /// Action interface
     void goalCB();
     void preemptCB();
-    void actionSuccess();
-    void actionAbort();
+    void actionSuccess(bool success, std::string message);
```

Comment:
Think about passing a string as a reference: const std::string& 
-> It is not necessary to change the string -> const
-> It is not necessary to create a copy of the string